docs/useGetList.md

```diff
@@ -118,3 +118,40 @@ const LatestNews = () => {
 ```
 
 The `data` will automatically update when a new record is created, or an existing record is updated or deleted.
+
+## TypeScript
+
+The `useGetList` hook accepts a generic parameter for the record type:
+
+```tsx
+import { useGetList } from 'react-admin';
+
+type Post = {
+    id: number;
+    title: string;
+};
+
+const LatestNews = () => {
+    // data is of type Post[]
```

Comment:
I'd put that comment lower in the snippet, when you actually use the data fields. And I'd rephrase it to something like

> TypeScript knows that `data` is of type Post[]